sycl/test-e2e/format.py

```diff
@@ -154,18 +154,26 @@ def execute(self, test, litConfig):
         if isinstance(script, lit.Test.Result):
             return script
 
-        devices_for_test = self.select_devices_for_test(test)
-        if not devices_for_test:
-            return lit.Test.Result(
-                lit.Test.UNSUPPORTED, "No supported devices to run the test on"
-            )
-
-        substitutions = lit.TestRunner.getDefaultSubstitutions(test, tmpDir, tmpBase)
+        devices_for_test = []
         triples = set()
-        for sycl_device in devices_for_test:
-            (backend, _) = sycl_device.split(":")
-            triples.add(get_triple(test, backend))
+        if "run-mode" not in test.config.available_features:
+            if "unsplit-mode" in test.requires or "TEMPORARY_DISABLED" in test.requires:
```

Comment:
This bit is for build-only mode, so that only the tests that are marked as `REQUIRES: unsplit-mode` or `REQUIRES: TEMPORARY_DISABLED` are reported as unsupported.

I guess it might be useful if I added a feature for run-only and build-only to avoid confusion in if statements like this

Reply:
ah sorry, i missed that we were checking the test requires

Reply:
I added the variable `test.config.split_mode` to hopefully make these types of if statements less confusing to read. (replaced `"run-mode" not in available_features` to `split_mode == "build-only"` )